### PR TITLE
feat: healthcheck + cors

### PR DESCRIPTION
**Pull request by PaulDremanovich**
Task: [Initial version of IPFS node](https://trello.com/c/d1iMesi5/386-initial-version-of-ipfs-node)

Completed items:
- Add `GET /node/info`. Include in the response: helia version, node version (from package.json), peerId, [multiaddresses](https://github.com/Adamant-im/ipfs-node/blob/2d1a6e199af329769b799098b7d7b5f39f86ba3a/src/index.js#L20), current unix time, storageSize, availableStorageSize
- Add Healthcheck endpoint `GET /node/health`. Include the UNIX timestamp in the response.
- Configure CORS with `config.default.json5`. Allow requests from: `*.adamant.im`, `adm.im`, `*.vercel.app`, `*.surge.sh`

### CODE REVIEW

**Comment by PaulDremanovich** [comment]
[Initial version of IPFS node](https://trello.com/c/d1iMesi5/386-initial-version-of-ipfs-node)

File: src/auto-peering.cron.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export const autoPeering = new CronJob(config.autoPeeringPeriod, () => {
   if (!started) {
     started = true
     autoPeeringHandler()
-      .catch(pino.logger.error)
+      .catch((err: Error) => pino.logger.error(`${err.message}\n${err.stack}`))
```
Comment (bludnic) [inline src/auto-peering.cron.ts]: `err: Error`: Is explicit typing required here?

File: src/index.ts [hunk 2]
```diff
@@ -399,6 +409,29 @@ app.post('/test', async (req, res) => {
...
+app.get('/node/health', async (req, res) => {
+  res.send({
+    timestamp: Date.now(),
+    heliaStatus: helia.libp2p.status
+  })
+})
+
+app.get('/node/info', async (req, res) => {
+  const { blockstoreSizeMb, datastoreSizeMb, availableSizeInMb } = getDiskUsageStats()
+  const heliaVersion = (packageJson.dependencies['helia'] || 'unknown').replaceAll('^', '')
```
Comment (bludnic) [inline src/index.ts]: Can you extract the Helia version using Helia API? See specification https://helia.io/

File: src/index.ts [hunk 3]
```diff
@@ -66,12 +68,20 @@ const ifs = unixfs(helia)
...
 
 const PORT = config.serverPort
 const app = express()
 
 app.use(pino)
 
+const corsOptions = {
+  origin: config.cors.originRegexps.map((item: string) => new RegExp(`^https?:\\/\\/${item}`)),
+  methods: ['GET', 'POST']
+}
```
Comment (bludnic) [inline src/index.ts]: What is the purpose of using RegExp?

Comment (PaulDremanovich) [reply src/index.ts]: To support wildcards such as *.adamant.im, adm.im, *.vercel.app, *.surge.sh

File: src/index.ts (hunk 2 above)
Comment (PaulDremanovich) [reply src/index.ts]: Unfortunately, I couldn't find such information in this doc. 
Therefore, I implemented getting the version this way.

File: src/index.ts (hunk 3 above)
Comment (bludnic) [reply src/index.ts]: Can you specify wildcards directly in the config?

File: src/index.ts [hunk 4]
```diff
@@ -66,12 +68,20 @@ const ifs = unixfs(helia)
...
+app.use(cors(corsOptions))
+app.options('*', cors())
```
Comment (bludnic) [inline src/index.ts]: I think this is not needed.

File: src/index.ts [hunk 5]
```diff
@@ -17,7 +17,9 @@ import type { PingService } from '@libp2p/ping'
 import { KadDHT } from '@libp2p/kad-dht'
 import { PeerId } from '@libp2p/interface'
 import { filestore } from './store.js'
-import config, { configFileName } from './config.js'
+import { config, configFileName, packageJson } from './config.js'
+import { getDiskUsageStats, diskUsageCron } from './disk-usage.cron.js'
+import cors from 'cors'
```
Comment (bludnic) [inline src/index.ts:22]: Install `@types/cors` to dev dependencies

Comment (PaulDremanovich) [reply src/index.ts:22]: Already install. Line 23 package.json

File: src/index.ts [hunk 6]
```diff
@@ -66,12 +68,19 @@ const ifs = unixfs(helia)
...
 const PORT = config.serverPort
 const app = express()
 
 app.use(pino)
 
+const corsOptions = {
+  origin: config.cors.originRegexps.map((item: string) => new RegExp(`^https?:\\/\\/${item}`)),
+  methods: ['GET', 'POST']
+}
+app.use(cors(corsOptions))
```
Comment (bludnic) [inline src/index.ts]: The `corsOptions` is not typed. Pass the object directly to `cors({...})`.

image

File: src/index.ts (hunk 2 above)
Comment (bludnic) [reply src/index.ts]: Then remove `heliaVersion` from nodeInfo response

**Review by bludnic — CHANGES_REQUESTED** [review]

File: src/disk-usage.cron.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    scan()
+      .catch((err) => pino.logger.error(`${err.message}\n${err.stack}`))
+      .finally(() => (started = false))
+  }
+})
+
+async function scan() {
+  pino.logger.info('[Cron] Running "diskUsage" cronjob.')
+  blockstoreSizeMb = (await dirSize(blockstorePath)) / 1048576
+  datastoreSizeMb = (await dirSize(datastorePath)) / 1048576
```
Comment (RealGoodProgrammer) [inline src/disk-usage.cron.ts]: Use const instead of magic numbers such as `const oneMb = 1048576`

**Review by RealGoodProgrammer — APPROVED** [review]

**Review by RealGoodProgrammer — APPROVED** [review]

**Review by bludnic — APPROVED** [review]

**Pull request merged**